src/sentry/models/featureadoption.py

```diff
@@ -0,0 +1,149 @@
+from __future__ import absolute_import
+
+from django.core.cache import cache
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.db.models import (
+    BaseManager,
+    FlexibleForeignKey,
+    Model,
+    sane_repr
+)
+
+
+class Feature(object):
+    """
+    prerequesites: list of feature_id's that are requirements or tuples (which are ORs)
+    """
+    def __init__(self, slug, name, location, prerequisite=None):
+        """
+        Args:
+            slug(str): unique
+            name(str): Human readable
+            type(str): what kind of feature: language, framework, code, webui, admin
+            prerequesite(list): list of feature slugs or tuples. Tuples indicate OR relationships.
+        """
+        self.slug = slug
+        self.name = name
+        self.location = location
+        self.prerequisite = prerequisite
+
+    def __repr__(self):
+        return self.name
+
+features = [
```

Comment:
The fact that this is a magic, mutable global that is used outside of this module scares me. We should implement a `FeatureManager` of some sort to do proper lookups and whatnot here.

This can also help us not use things like, `[f.slug for f in features if f.location == 'language']`

It actually took me a few minutes to realize what was happening there since `features` didn't indicate anything about it being this global variable. :)